src/controllers/hid/hidioreport.cpp

```diff
@@ -0,0 +1,64 @@
+#include "controllers/hid/hidioreport.h"
+
+#include <hidapi.h>
+
+#include "controllers/defs_controllers.h"
+#include "controllers/hid/legacyhidcontrollermappingfilehandler.h"
+#include "util/string.h"
+#include "util/time.h"
+#include "util/trace.h"
+
+namespace {
+constexpr int kReportIdSize = 1;
+constexpr int kMaxHidErrorMessageSize = 512;
+QString loggingCategoryPrefix(const QString& deviceName) {
+    return QStringLiteral("controller.") +
+            RuntimeLoggingCategory::removeInvalidCharsFromCategory(deviceName.toLower());
+}
+} // namespace
+
+HidIoReport::HidIoReport(const unsigned char& reportId,
+        hid_device* device,
+        const mixxx::hid::DeviceInfo&& deviceInfo)
+        : m_reportId(reportId),
+          m_logOutput(loggingCategoryPrefix(deviceInfo.formatName()) + QStringLiteral(".output")),
+          m_pHidDevice(device),
+          m_deviceInfo(std::move(deviceInfo)) {
```

Comment:
I believe you added std::move to the caller -- it doesn't help to sprinkle std::moves everywhere :).  Why move?  Just take a const& and make a copy.

Reply:
(Now I see this is the report, not the thread -- but the issue still stands)